examples/browser/src/components.ts

```diff
@@ -22,12 +22,12 @@ export function add(lhs: PartialVec2, rhs: PartialVec2): Vec2 {
 
 /** Subtracts two 2D vectors. */
 export function sub(lhs: PartialVec2, rhs: PartialVec2): Vec2 {
-  return { x: (lhs.x ?? 0) - (rhs.x ?? 0), y: (lhs.y ?? 0) - (rhs.y ?? 0) };
+  return add(lhs, scale(rhs, -1));
```

Comment:
Yeah we could do this, my idea was that subtraction is so fundamental that it ought to be as fast as possible, so I'd actually lean towards keeping it as is (though perhaps we'd want to default the parameters to `{ x = 0, y = 0 }` to make it less verbose).